### PR TITLE
Add release script

### PR DESCRIPTION
**Pull request by ltrung**
**Description of changes:**
Add a new release script and GitHub action to publish to NPM for each release.
**Testing**

1. Have you successfully run `npm run build:release` locally? Yes
2. How did you test these changes? Manually running the script
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? N/A
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: script/release [hunk 1]
```diff
@@ -99,6 +96,6 @@ x = STDIN.gets
 exit(1) unless x.strip == 'yes'
 puts
 
-verbose("git push origin")
-verbose("git push origin -f #{tag}")
-verbose("git push origin -f #{semvertag}")
+verbose("git push origin HEAD:release --follow-tags -f")
+
+puts "Remember to tag the release with #{semvertag}"
```
Comment (michhyun1) [inline script/release:101]: Should this script create the PR with the tag?

File: script/release [hunk 2]
```diff
@@ -86,11 +86,8 @@ verbose("git tag -d #{tag} || true")
 verbose("git tag #{tag}")
 verbose("git tag #{semvertag} -m \"Release #{semvertag}.\"")
```
Comment (michhyun1) [inline script/release:88]: why is there two tags?

File: script/release (hunk 1 above)
Comment (ltrung) [reply script/release:101]: I can remove it. Just a reminder to use which tag to tag the release when drafting new release.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - name: NPM Install
+        run: npm install
+      - name: NPM run build
+        run: npm run build
+      - name: Publish to NPM
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
```
[the remark this answers is not in the document]
Comment (ltrung) [reply .github/workflows/publish.yml:25]: I will shadow you on Monday and will add them then :)

File: script/release (hunk 2 above)
Comment (ltrung) [reply script/release:88]: This is from the old publish script. We already tag twice with the old one. Richard and Andrew said we need to make sure nobody uses them before removing one of them so i leave it there.

**Review by michhyun1 — APPROVED** [review]

**Pull request merged**